AK/UFixedBigInt.h

```diff
@@ -0,0 +1,544 @@
+
+    template<Unsigned U>
+    constexpr R&& operator+(U other) const
+    {
+        bool carry; // unused
+        return addc(other, carry);
+    }
+    template<Unsigned U>
+    constexpr R&& operator-(U other) const
+    {
+        bool carry; // unused
+        return subc(other, carry);
+    }
+
+    template<Unsigned U>
+    constexpr R& operator+=(U other)
+    {
+        bool carry;
+        *this = addc(other, carry);
+        return *this;
+    }
+    template<Unsigned U>
+    constexpr R& operator-=(U other)
+    {
+        bool carry;
+        *this = subc(other, carry);
+        return *this;
+    }
+
+    constexpr R&& operator++()
+    {
+        // x++
+        auto old = *this;
+        *this += 1;
+        return old;
+    }
+    constexpr R&& operator++(int)
+    {
+        // ++x
+        *this += 1;
+        return *this;
+    }
+    constexpr R&& operator--()
+    {
+        // x--
+        auto old = *this;
+        *this -= 1;
+        return old;
+    }
+    constexpr R&& operator--(int)
+    {
+        // --x
+        *this -= 1;
+        return *this;
+    }
+
+    // FIXME: following maybe really inefficient
+    // FIXME: less restraints on this;
+    template<Unsigned U>
+    requires(sizeof(R) >= sizeof(U)) constexpr R&& div_mod(const U& other, U& remainder) const
+    {
+        // FIXME: Is this correct?
+        // FIXME: Is there a better way to raise a division by 0?
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdiv-by-zero"
+        if (!other)
+            [[maybe_unused]]
+            auto x = 1 / 0;
+#pragma GCC diagnostic pop
+
+        remainder = 0;
+        R quotient = 0;
+        for (ssize_t i = sizeof(R) * 8; i > 0; --i) {
+            remainder <<= 1;
+            remainder |= (*this >> i) & 1;
+            if (remainder >= other) {
+                remainder -= other;
+                quotient |= 1 < i;
+            }
+        }
+
+        return quotient;
+    }
+
+    template<Unsigned U>
+    constexpr R&& operator*(U other) const
+    {
+        // FIXME: Is this correct?
+        R res = 0;
+        for (; other; other >>= 1) {
+            res << 1;
+            if (other & 1)
+                res += *this;
+        }
+        return res;
+    }
+
+    template<Unsigned U>
+    constexpr R&& operator/(const U& other) const
+    {
+        R mod { 0 };
+        return div_mod(other, mod);
+    }
+    template<Unsigned U>
+    constexpr R&& operator%(const U& other) const
+    {
+        R res { 0 };
+        div_mod(other, res);
+        return res;
+    }
+
+    template<Unsigned U>
+    constexpr R& operator*=(const U& other)
+    {
+        *this = *this * other;
+        return *this;
+    }
+    template<Unsigned U>
+    constexpr R& operator/=(const U& other)
+    {
+        *this = *this * other;
+        return *this;
+    }
+    template<Unsigned U>
+    constexpr R& operator%=(const U& other)
+    {
+        *this = *this * other;
+        return *this;
+    }
```

Comment:
Some copy-paste shenanigans here: these should be `*this * other`, `*this / other` and `*this % other` respectively

Reply:
ups..., fixed that, and will be pushed once I get the requires for addc working correctly